kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/RealtimePlanning.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.optaplanner;
+
+import java.util.List;
+
+import org.optaplanner.core.api.solver.event.SolverEventListener;
+import org.optaplanner.core.impl.solver.ProblemFactChange;
+
+/**
+ * Provides integration for OptaPlanner to solve planning problems real time 
+ * based on support of adding problem facts.
+ *
+ */
+public interface RealtimePlanning {
+
+    /**
+     * Invoked at the initialization phase of OptaPlanner Extension to identify planning that are active.
+     * @return returns true if this planning was successfully activated, false otherwise. In case of false returned 
+     * this instance of planning will not be called anymore.
+     */
+    boolean init();
```

Comment:
If something fails, it will probably always thrown a RuntimeException (in the fail fast philosophy).
Why would we ever want to return false? I soft suggest to make method this return void.